src/engine/strat_engine/backstore/metadata/bda.rs

```diff
@@ -283,7 +300,7 @@ impl StaticHeader {
     /// Return an error if the sigblocks differ in some unaccountable way.
     /// Returns an error if a write intended to repair an ill-formed,
     /// unreadable, or stale signature block failed.
-    fn setup<F>(f: &mut F) -> StratisResult<Option<StaticHeader>>
+    fn setup<F>(f: &mut F) -> StratisResult<Option<SetupResult>>
```

Comment:
Note that your change also requires a change to the header comment, since the method no longer just returns an error if the repair failed.